packages/cypher-builder/src/utils/escape.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const ESCAPE_SYMBOL_REGEX = /`/g;
+
+export function escapeLabel(label: string): string {
+    // TODO: only escape when needed
+    return escapeString(label);
+}
+
+export function escapeType(type: string): string {
+    const normalizedStr = normalizeString(type);
+    if (needsEscape(normalizedStr)) {
+        return escapeString(normalizedStr);
+    }
+    return type;
+}
+
+export function escapeProperty(propName: string): string {
+    const normalizedStr = normalizeString(propName);
+    if (needsEscape(normalizedStr)) {
+        return escapeString(normalizedStr);
+    }
+    return propName;
+}
+
+function normalizeString(str: string): string {
+    return str.replace(/\\u0060/g, "`");
+}
+
+function needsEscape(str: string): boolean {
+    const validCharacter = /^[a-z0-9_]*$/i;
```

Comment:
We can refine this in the future, but this is a really basic regex for the basic cases, for anything non-obvious, better to rely on escaping I think, just to be safe. :thinking: 

In fact, maybe we should exclude numbers, as `(A:1)` is not valid without escaping

Reply:
Your call! If we choose to go with it, we may change the changes note from `patch` to `minor` as for some users it will be a breaking change.
BTW,  I believe that the only way to produce this:
`(this0)-[this1:ACTED_IN|DIRECTED]->(this2)`
Is by defining:
```
const a = new Cypher.Node();
const b = new Cypher.Node();
const rel = new Cypher.Relationship({
      type: "ACTED_IN|DIRECTED",
});
new Cypher.Pattern(a).related(rel).to(b)
```
If this is the case, correct me if I'm wrong here, we will prevent users from abusing it, although perhaps we should think of a proper solution here.

Reply:
I'll disable relationship escaping and create a new PR on 4.0.0 to avoid breaking changes